### PR TITLE
Refactor: Misc structure, typing, build, etc

### LINKED ISSUE #10: [FEATURE] Build both cjs and esnext

### LINKED ISSUE #11: [FEATURE] Export type definitions

### PR DESCRIPTION
**Pull request by sindrekjr**
## Description
Various changed to build process, type declarations, and project structure. 

- Added needlessly complex build scripts to build for both cjs and esnext without duplicating too many files. 
- No longer declare types globally; instead import where needed. 
- Package now exports certain types.
- Moved `SentenceFactory` to module/folder `sentence` for less blobbing.
- Removed JSONs. 

## Checklist
- [x] Changelog has been updated to reflect changes.

### CODE REVIEW

File: src/defaults/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (RavnIv) [inline src/defaults/index.ts:1]: Why is this needed?

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export interface Configuration {
+  options?: Options;
+  templates?: Template[];
+  vocabulary?: Vocabulary;
+}
+
+export interface DefinitelyOptions {
```
Comment (RavnIv) [inline types/index.d.ts:7]: The naming here seems maybe a bit weird 😛 
Maybe something like `ResolvedOptions` instead?

Comment (sindrekjr) [reply types/index.d.ts:7]: Yeah, it's a bit weird, but between `DefinitelyOptions` and something like `ResolvedOptions` I figured I'd go for "definitely" as a silly reference to "DefinitelyTyped", which is a ts thing. :P 

`ResolvedOptions` does sound a lot better though.

File: src/defaults/index.ts (hunk 1 above)
Comment (sindrekjr) [reply src/defaults/index.ts:1]: Since eslint isn't quite fully compatible with typescript, it doesn't understand how imported types are used, so it will throw errors here even if you've used the types in your code. What I've done here is admittedly a kinda messy workaround, but not sure what else I should do when I want to keep eslint and the `no-unused-vars` rule. 

I'm considering trying to switch from eslint to tslint though.

**Review by RavnIv — APPROVED** [review]
